src/main/scala/scalaz/stream/wye.scala

```diff
@@ -836,9 +836,7 @@ object wye {
         }
       })(S)
 
-      repeatEval(Task.async[Seq[O]] { cb => a ! Get(cb) })
-      .flatMap(emitAll)
-      .onComplete(eval_(Task.async[Unit](cb => a ! DownDone(cb))))
+      repeatEval(Task.async[Seq[O]] { cb => a ! Get(cb) }) onHalt { _.asHalt } flatMap emitAll onComplete eval_(Task.async[Unit](cb => a ! DownDone(cb)))
```

Comment:
@djspiewak generally, I am not really sure if I am following your arguments, but really having that long lines make reasoning about the code very hard. 

However, I think we have do adopt one code style and keep it through the library, consistent. If we would change code style to your proposal, I have no problem with that, but I would like to keep line with at reasonable level and all new PRs to this library shall follow it, as currently, we use dot notation almost everywhere. 

I think this is something we need collective agreement on what do you think? 

Just for curiosity whats wrong with writing your code as below?

```scala

repeatEval(Task.async[Seq[O]] ( cb => a ! Get(cb) )) 
.onHalt(_.asHalt) 
.flatMap(emitAll)
.onComplete(eval_(Task.async[Unit](cb => a ! DownDone(cb))))

```